### ISSUE TITLE
Using regex with Elastica

### ISSUE THREAD

**Issue by Doweig** [ruflin/Elastica#498]
I have been using elastisearch trough elastica recently and I am having problems using the regexp.

Here is my problem:

I need to find all instances of my "Foo" entity with the "name" NOT begin by a letter.

``` PHP
class Foo
{
    protected $name; // example: "1&1", "OVH", "Aliyun" ...
}
```

According to elasticsearch documentation, there is a "regexp" filter but I couldn't find the implementation in elastica.

I have tried:

``` PHP
\Elastica\Query\Prefix(); //Apparently it doesn't handle the regexp
```

From the documentation I could get:

``` PHP
$reg = '@&~(foo.+)'; # anything except string beginning with "foo"
```

Tweaked to:

``` PHP
$query = new \Elastica\Query\Prefix();
$query->setPrefix('name',  '@&~([a-z].+)');
```

Doesn't work

Basically I can't find what type of query class I should use from Elastica.

Thank you for you help !
